extensions/amp-hulu/0.1/amp-hulu.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+
+class AmpHulu extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  preconnectCallback() {
+    this.preconnect.url('https://secure.hulu.com/dash/mobile_embed.html');
```

Comment:
@unbug let's do @jridgewell's suggestion for preload but you need to have the `eid=` appended to the Url before preloading. I suggest the following:

1- introduce a new instance variable `this.eid_`  in the cosntrcutor.
2- in `buildCallback` do the `this.eid_ = user().assert(this.element.getAttribute('data-eid'),
         'The data-eid attribute is required for <amp-hulu> %s',
         this.element);` instead of in the layoutCallback as it does now.
3- Create a new `getVideoIframeSrc_()` which returns the Url with correct eid query string appended.
4- Use `getVideoIframeSrc_()`  in `preconnectCallback` and `layoutCallback`